source/extensions/filters/http/ratelimit/ratelimit_headers.cc

```diff
@@ -0,0 +1,91 @@
+#include "extensions/filters/http/ratelimit/ratelimit_headers.h"
+
+#include "common/http/header_map_impl.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace RateLimitFilter {
+
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_limit_handle(Http::CustomHeaders::get().XRateLimitLimit);
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_remaining_handle(Http::CustomHeaders::get().XRateLimitRemaining);
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::ResponseHeaders>
+    x_rate_limit_reset_handle(Http::CustomHeaders::get().XRateLimitReset);
+
+Http::ResponseHeaderMapPtr RateLimitHeaders::create(
+    Filters::Common::RateLimit::DescriptorStatusListPtr&& descriptor_statuses) {
+  Http::ResponseHeaderMapPtr result = Http::ResponseHeaderMapImpl::create();
+  if (!descriptor_statuses || descriptor_statuses->empty()) {
+    descriptor_statuses = nullptr;
+    return result;
+  }
+
+  absl::optional<envoy::service::ratelimit::v3::RateLimitResponse_DescriptorStatus>
+      min_remaining_limit_status;
+  std::string quota_policy;
+  for (auto&& status : *descriptor_statuses) {
+    if (!status.has_current_limit()) {
+      continue;
+    }
+    if (!min_remaining_limit_status ||
+        status.limit_remaining() < min_remaining_limit_status.value().limit_remaining()) {
+      min_remaining_limit_status.emplace(status);
+    }
+    uint32_t window = convertRateLimitUnit(status.current_limit().unit());
+    // Constructing the quota-policy per RFC
+    // https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html#rfc.section.3.1
+    // Example of the result: `, 10;window=1;name="per-ip", 1000;window=3600`
+    if (window) {
+      // For each descriptor status append `<LIMIT>;window=<WINDOW_IN_SECONDS>`
+      absl::StrAppend(&quota_policy,
+                      fmt::format(", {:d};{:s}={:d}", status.current_limit().requests_per_unit(),
```

Comment:
nit: instead of using fmt::format here I think you can just pass multiple arguments to StrAppend and do this more efficiently. I would take a look at https://abseil.io/docs/cpp/guides/strings.